sensors/simsensor_common/event_queue.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Queue for buffering sensor event data from reader to simsensors
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "event_queue.h"
+
+#include <stdlib.h>
+
+
+struct event_queue_t {
+	sensor_event_t *queue;
+	unsigned int queue_sz;
+	unsigned int act_elem_cnt;
+
+	sensor_event_t *head;
+	sensor_event_t *tail;
+};
+
+
+event_queue_t *eventQueue_init(unsigned int elem_cnt)
+{
+	event_queue_t *result;
+
+	if (elem_cnt == 0) {
+		return NULL;
+	}
+
+	result = malloc(sizeof(event_queue_t));
```

Comment:
The `event_queue_t` structure can be passed as an argument (pointer). No need to allocate it dynamically